--- /dev/null
+++ b/build.gradle.kts
@@ -1,0 +_,77 @@
+plugins {
+    java
+    id("xyz.jpenilla.run-paper") version "2.2.4"
+    id("net.minecrell.plugin-yml.paper") version "0.6.0"
+    id("io.github.goooler.shadow") version "8.1.7"
+    `maven-publish`
+    id("io.papermc.hangar-publish-plugin") version "0.1.2"
+    id("com.modrinth.minotaur") version "2.+"
+}
+
+repositories {
+    mavenCentral()
+    maven("https://repo.papermc.io/repository/maven-public/")
+}
+
+dependencies {
+    compileOnly("io.papermc.paper:paper-api:1.20.4-R0.1-SNAPSHOT")
+    implementation("org.bstats:bstats-bukkit:3.0.2")
+    implementation("org.incendo:cloud-core:2.0.0-beta.5")
+    implementation("org.incendo:cloud-annotations:2.0.0-beta.5")
+    implementation("org.incendo:cloud-paper:2.0.0-beta.5")
+
+    implementation(project(":coincoffer-api"))
+}
+
+group = "dev.onelitefeather.plugin.coincoffer"
+description = "CoinCoffer"
+java.sourceCompatibility = JavaVersion.VERSION_17
+
+tasks {
+    compileJava {
+        options.encoding = "UTF-8"
+    }
+    javadoc {
+        options.encoding = "UTF-8"
+    }
+    runServer {
+        minecraftVersion("1.20.4")
+    }
+    shadowJar {
+        relocate("org.bstats", "dev.onelitefeather.plugin.coincoffer.org.bstats")
+    }
+}
+
+paper {
+    name = "CoinCoffer"
+    main = "net.milkbowl.vault.Vault"
+    foliaSupported = true
+    apiVersion = "1.19"
+    authors = listOf("TheMeinerLP", "OneLiteFeatherNET")
+    provides = listOf("Vault")
+}
+
+modrinth {
+    token.set(System.getenv("MODRINTH_TOKEN"))
+    versionNumber.set(rootProject.version.toString())
+    projectId.set("9KWfS9j4")
+    versionType.set("release")
+    uploadFile.set(tasks.shadowJar)
+    gameVersions.addAll(listOf("1.20.5","1.20.4","1.20.3", "1.20.2", "1.20.1", "1.20", "1.19.4", "1.19.3", "1.19.2", "1.19.1", "1.19"))
+    loaders.add("paper")
+}
+
+hangarPublish {
+    publications.register("CoinCoffer") {
+        version.set(rootProject.version.toString())
+        channel.set("Release")
+        apiKey.set(System.getenv("HANGAR_KEY"))
+        id.set("CoinCoffer")
+        platforms {
+            paper {
+                jar.set(tasks.shadowJar.flatMap { it.archiveFile })
+                platformVersions.set(listOf("1.20.5","1.20.4","1.20.3", "1.20.2", "1.20.1", "1.20", "1.19.4", "1.19.3", "1.19.2", "1.19.1", "1.19"))
+            }
+        }
+    }
+}
