--- /dev/null
+++ b/build.gradle.kts
@@ -1,0 +_,70 @@
+plugins {
+    java
+}
+
+group = "net.milkbowl.vault"
+version = "1.7.3"
+description = "Vault"
+java.sourceCompatibility = JavaVersion.VERSION_17
+
+
+repositories {
+    mavenLocal()
+    mavenCentral()
+    maven {
+        url = uri("https://nexus.hc.to/content/repositories/pub_releases/")
+
+        // isAllowInsecureProtocol = true
+    }
+
+    maven {
+        url = uri("https://hub.spigotmc.org/nexus/content/groups/public/")
+    }
+
+    maven {
+        url = uri("https://dev.escapecraft.com/maven")
+
+        // isAllowInsecureProtocol = true
+    }
+
+    maven {
+        url = uri("https://repo.codemc.org/repository/maven-public")
+    }
+
+    maven {
+        url = uri("https://repo.maven.apache.org/maven2/")
+    }
+}
+
+dependencies {
+    /*api(libs.org.bukkit.bukkit)
+    api(libs.net.milkbowl.vault.vaultapi)
+    api(libs.org.bstats.bstats.bukkit)*/
+    /*system(libs.de.hydrox.bukkit.droxperms)
+    system(libs.net.krinsoft.privileges)
+    system(libs.de.bananaco.permissions.permissions.bpermissions)
+    system(libs.com.github.sebc722.xperms)
+    system(libs.org.anjocaido.groupmanager.groupmanager.groupmanager)
+    system(libs.net.thedgtl.ichat.ichat.ichat)
+    system(libs.net.d3gn.miraclem4n.mchat.mchat.mchat)
+    system(libs.`in`.mdev.mchatsuite)
+    system(libs.com.nijikokun.bukkit.permissions.permissions.permissions)
+    system(libs.com.platymuus.bukkit.permissions.permissionsplugin.permissionsbukkit)
+    system(libs.ru.tehkode.permissionsex)
+    system(libs.de.bananaco.permissions.permissions.bpermissions2)
+    system(libs.com.dthielke.starburst.starburst)
+    system(libs.net.crystalyx.bukkit.simplyperms.simplyperms)
+    system(libs.com.github.zathrus.writer.commandsex.commandsex)
+    system(libs.net.ae97.totalpermissions)
+    system(libs.ru.simsonic.rscpermissions)
+    system(libs.kperms.kperms)
+    system(libs.com.overmc.overpermissions)*/
+}
+
+tasks.withType<JavaCompile>() {
+    options.encoding = "UTF-8"
+}
+
+tasks.withType<Javadoc>() {
+    options.encoding = "UTF-8"
+}
